src/services/TaskRepository.ts

```diff
@@ -0,0 +1,114 @@
+//import { Tasks } from "../models/ArrayModel";
+export class TaskRepo {
+    getTasks(completion:boolean, sortBy:string, sortOrder:string){
+        var queriedTasks:Task[]
+
+        switch (completion)
+        {
+            case true:
+                queriedTasks = Tasks.filter(Task => Task.completed == true);
+                break;
+            case false:
+                queriedTasks = Tasks.filter(Task => Task.completed == false);
+                break;
+            default:
+                queriedTasks = Tasks;
+        }
+
+        // Selects case based upon what query parameter was entered for sorting. 
+        switch (sortBy)
+        {
+            case 'createdAt':
+                
+                if (sortOrder == 'asc')
+                {
+                    queriedTasks.sort((a,b) => +new Date(a.createdAt) - +new Date(b.createdAt));
+                }
+                else if (sortOrder == 'desc')
+                {
+                    queriedTasks.sort((a,b) => +new Date(b.createdAt) - +new Date(a.createdAt));
+                }
+                break;
+
+            case 'dueDate':
+                if (sortOrder == 'asc')
+                {
+                    queriedTasks.sort((a,b) => +new Date(a.dueDate) - +new Date(b.dueDate));
+                }
+                else if (sortOrder == 'desc')
+                {
+                    queriedTasks.sort((a,b) => +new Date(b.dueDate) - +new Date(a.dueDate));
+                }
+                break;
+        }
+        return queriedTasks;
+    }
+
+    fetchTask(id:number){
+        return Tasks[Tasks.findIndex(Task => Task.id == id)]
```

Comment:
Nice find on `findIndex` I generally do `indexOf`

My question though is, what if a task with a `id` value is not found, what happens then?